win32/src/uuid_win32.c

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "rpc.h"
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/xlogging.h"
+
+#include "c_pal/uuid.h"
+
+static int is_UUID_T_and_UUID_same_size[sizeof(UUID_T) == sizeof(UUID)]; /*just a sanity check*/
+
+/*these are interface requirements*/
+/*Codes_SRS_UUID_02_001: [ If destination is NULL then uuid_produce shall fail and return a non-NULL value. ]*/
+/*Codes_SRS_UUID_02_002: [ uuid_produce shall generate in destination the representation of a UUID (as per RFC 4122). ]*/
+/*Codes_SRS_UUID_02_004: [ uuid_produce shall succeed and return 0. ]*/

```

Comment:
nit: we could still put these next to the code that implements this

Reply:
Frankly, here's a bit of a non-consistency as we treat these "interface" requirements.

I would argue that the implementation ("codes") of these requirements is actually other requirements.  (not code).

As in "SRS_UUID_WIN32_02_001" implements  "SRS_UUID_02_001". 

And I am not even sure about testing the interface requirements, maybe some int tests would "test" these interface requirements.

I guess this is why I wrote in line 14 the comment with "these are interface requirements", because they are scattered in this file (and in the linux one) too.

Reply:
I moved them

Reply:
Yeah I do think that the interface requirements should be tested in the INT tests in the interface folder probably